msprime/mutations.py

```diff
@@ -199,6 +199,308 @@ def __init__(self, state_independent=False):
+    :param int lo: lower bound on repeat number
+    :param int hi: upper bound on repeat number
+    :param list[float] root_distribution: An array of float values of length hi-lo+1
+        which should sum to 1, and give the probabilities used to choose
+        the ancestral state of each microsatellite.
+        Defaults to the stationary distribution of the model.
+
+    """
+
+    def __init__(
+        self,
+        s=0,
+        u=0.5,
+        v=0.0,
+        p=1.0,
+        m=1.0,
+        lo=2,
+        hi=50,
+        root_distribution=None,
+    ):
+        if np.isinf(lo):
+            raise ValueError("lo is infinite")
+        if np.isinf(hi):
+            raise ValueError("hi is infinite")
+        if lo > hi:
+            raise ValueError("lo > hi")
+        if not lo >= 0:
+            raise ValueError("lo must be >= 0")
+        if s <= -1 / (hi - lo + 1):
+            raise ValueError("s must be > -1/(hi-lo+1)")
+        if u < 0 or u > 1:
+            raise ValueError("u must be between 0 and 1 inclusive")
+        if type(lo) == float:
+            lo = int(lo)
+        if type(hi) == float:
+            hi = int(hi)
+        alleles = [str(int(x)) for x in range(lo, hi + 1)]
+        transition_matrix = general_microsat_rate_matrix(s, u, v, p, m, lo, hi)
+        if root_distribution is None:
+            # solve for stationary distribution
+            S, U = np.linalg.eig(transition_matrix.T)
+            U = np.real_if_close(U, tol=1)
+            stationary = np.array(U[:, np.where(np.abs(S - 1.0) < 1e-8)[0][0]])
+            stationary = stationary / np.sum(stationary)
+            root_distribution = stationary
+        super().__init__(alleles, root_distribution, transition_matrix)
+
+
+class SMM(MicrosatMutationModel):
+    """
+    Stepwise mutation model (Ohta and Kimura, 1978),
+    for microsatellite repeat number.
+
+    This is a :class:`.MicrosatMutationModel` with alleles ``[lo, .. , hi]``,
+    a root distribution (stationary distribution by default),
+    and a transition matrix that allows only mutations that
+    change the number of repeats by +/- 1: concretely, if the mutation
+    rate is ``m``, then an allele ``k`` mutates to ``k+1`` and ``k-1``
+    at rate ``m/2`` each,
+    except that mutations above ``hi`` or below ``lo`` result in no change.
+
+    Note that the values of ``lo`` and ``hi`` are in units of repeat number,
+    and are unrelated to repeat length
+    (e.g., the repeat itself can be dinucleotide, tri-nucleotide, etcetera).
+
+    :param int lo: lower bound of the microsatellite repeat copy number
+    :param int hi: upper bound of the microsatellite repeat copy number
+    :param list[float] root_distribution: An array of float values of length hi-lo+1
+        which should sum to 1. These values are used to determine the ancestral state of
+        each microsatellite. Defaults to the value of stationary distribution
+        of the model
```

Comment:
This is what I meant here:
```suggestion
        which should sum to 1, and give the probabilities used to choose
        the ancestral state of each microsatellite.
        Defaults to the value of stationary distribution
        of the model.
```